### PR TITLE
fix: Use openapi-types 3.1 document type to support null schema types

### PR DESCRIPTION
**Pull request by nidhi-kala**
Fixes https://github.com/fastify/fastify-swagger/issues/705 

```
Type '"null"' is not assignable to type 'NonArraySchemaObjectType | undefined'.
```

By using `OpenAPIV3_1.Document` instead of `OpenAPIV3.Document` as type for Fastify dynamic swagger options, openapi config.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { FastifyPluginCallback, FastifySchema, onRequestHookHandler, preHandlerHookHandler } from 'fastify';
-import { OpenAPI, OpenAPIV2, OpenAPIV3 } from 'openapi-types';
+import { OpenAPI, OpenAPIV2, OpenAPIV3, OpenAPIV3_1 } from 'openapi-types';
```
Comment (Uzlopak) [inline index.d.ts:2]: If we use OpenAPIV3_1 then we dont need OpenAPIV3. So modify in line the namespace merging by changing `namespace OpenAPIV3` to `namespace OpenAPIV3_1`

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I guess it is OK. But see my other remark.

File: index.d.ts (hunk 1 above)
Comment (nidhi-kala) [reply index.d.ts]: Okay, I updated it.

**Review by climba03003 — CHANGES_REQUESTED** [review]
Please use `OpenAPIV3 | OpenAPIV3_1` syntax. OpenAPI 3.1 is not a superset of OpenAPI 3.0

**Review by climba03003 — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**